progress_bar_circular.go

```diff
@@ -0,0 +1,204 @@
+package unichart
+
+import (
+	"io"
+	"math"
+
+	"github.com/unidoc/unichart/mathutil"
+	"github.com/unidoc/unichart/render"
+	"github.com/unidoc/unipdf/v3/model"
+)
+
+// CircularProgressBar is a component that will render circular progress bar component.
+type CircularProgressBar struct {
+	BackgroundStyle render.Style
+	ForegroundStyle render.Style
+	LabelStyle      render.Style
+	ColorPalette    render.ColorPalette
+
+	// If the value is true then the progress bar would rendered counter clockwise.
+	Reversed bool
+
+	// Size of the progress bar in width and height.
+	size int
+	dpi  float64
+
+	// Progress values which should be between 0.0 - 1.0.
+	progress float64
+
+	// Label to be displayed in the center of the chart.
+	label string
+}
+
+// SetProgress set the progress that will represented by this chart.
+// Expected value should be float number between 0.0 - 1.0.
+func (cp *CircularProgressBar) SetProgress(progress float64) {
+	cp.progress = math.Max(0.0, math.Min(progress, 1.0))
+}
+
+// GetProgress returns the progress represented by this chart.
+// Returned value should be a float number between 0.0 - 1.0.
+func (cp *CircularProgressBar) GetProgress() float64 {
+	return cp.progress
+}
+
+// DPI returns the DPI of the progress bar.
+func (cp *CircularProgressBar) DPI() float64 {
+	if cp.dpi == 0 {
+		return defaultDPI
+	}
+	return cp.dpi
+}
+
+// SetDPI sets the DPI for the progrss bar.
+func (cp *CircularProgressBar) SetDPI(dpi float64) {
+	cp.dpi = dpi
+}
+
+// Width returns the chart size or the default value.
+func (cp *CircularProgressBar) Size() int {
+	if cp.size == 0 {
+		return defaultChartWidth
+	}
+	return cp.size
+}
+
+// SetWidth sets the chart size.
+func (cp *CircularProgressBar) SetSize(size int) {
+	cp.size = size
+}
+
+func (cp *CircularProgressBar) SetLabel(label string) {
+	cp.label = label
+}
+
+func (cp *CircularProgressBar) GetLabel() string {
+	return cp.label
+}
+
+// Width returns the chart width.
+func (cp *CircularProgressBar) Width() int {
+	return cp.Size()
+}
+
+// SetWidth sets the chart width.
+func (cp *CircularProgressBar) SetWidth(width int) {
```

Comment:
Empty function? or this is needed to implements interface?

Reply:
yes, these empty functions is needed to fulfill the interface requirements. I'll add more info in the comments later.